tests/testthat/test-target_sda.R

```diff
@@ -616,3 +616,39 @@ test_that("doesn't output NAs if ald and scenario years are misaligned (#307,
     any(is.na(out$emission_factor_value))
   )
 })
+
+
+test_that("output useful error message when emission_factor is not of type double (#224)", {
+      matched <- tibble::tribble(
+       ~id_loan, ~loan_size_outstanding, ~loan_size_outstanding_currency, ~loan_size_credit_limit, ~loan_size_credit_limit_currency, ~id_2dii,            ~level, ~score,      ~sector,      ~name_ald, ~sector_ald,
+      "L162",                      1,                           "EUR",                       2,                            "EUR",    "UP1", "ultimate_parent",      1, "automotive", "shaanxi auto",    "cement"
+    )
+    bad_ald = tibble::tribble(
+      ~name_company,  ~sector, ~technology, ~year, ~production, ~emission_factor, ~plant_location, ~is_ultimate_owner,
+      "shaanxi auto", "cement",    "cement",  2020,           1,                1,            "BF",               TRUE,
+      "shaanxi auto", "cement",    "cement",  2021,           1,                2,            "BF",               TRUE,
+      "shaanxi auto", "cement",    "cement",  2022,           1,                3,            "BF",               TRUE
+    )
+    co2_intensity_scenario <- tibble::tribble(
+      ~scenario,  ~sector,  ~region, ~year, ~emission_factor,           ~emission_factor_unit, ~scenario_source,
+      "b2ds", "cement", "global",  2020,              0.6, "tons of CO2 per ton of cement",      "demo_2020",
+      "b2ds", "cement", "global",  2050,              0.2, "tons of CO2 per ton of cement",      "demo_2020"
+    )
+    bad_ald <- ald %>%
+      mutate(
+        emission_factor = as.character(emission_factor)
+      )
+    out <- target_sda(
+      matched,
+      bad_ald,
+      co2_intensity_scenario
+    )
```

Comment:
You don't need to define the object `out` here, since you will call directly the function `target_sda()` within the function `expect_error`. Also note, in general, if you are expecting a function call to error, you should always be wrapping it in an `expect_error` call otherwise you will have tests failing that you expect to pass.